configs/components/ruby-selinux.rb

```diff
@@ -20,10 +19,16 @@
   pkg.build_requires "libsepol-devel"
   pkg.build_requires "libselinux-devel"
 
+  ruby = "#{settings[:bindir]}/ruby -rrbconfig"
+  if platform.is_cross_compiled_linux?
+    pkg.environment "RUBY" => settings[:host_ruby]
+    ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig.rb"
+  end
+
   pkg.build do
-    ["export RUBYHDRDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"rubyhdrdir\"]')",
-     "export VENDORARCHDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"vendorarchdir\"]')",
-     "export ARCHDIR=$${RUBYHDRDIR}/$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG[\"arch\"]')",
+    ["export RUBYHDRDIR=$(shell #{ruby} -e 'puts RbConfig::CONFIG[\"rubyhdrdir\"]')",
```

Comment:
This is more of a personal nitpick so it's nothing I'd block over. But when you've got multiple kinds of nested quotations going on like this, consider the humble `%()`:

```ruby
%(export RUBYHDRDIR=$(shell #{settings[:bindir]}/ruby -rrbconfig -e 'puts RbConfig::CONFIG["rubyhdrdir"]'))
```

A little less escaping, a slightly cleaner string (I think).